rootchain/sampool/pool_test.go

```diff
@@ -0,0 +1,356 @@
+package sampool
+
+import (
+	"errors"
+	"github.com/0xPolygon/polygon-edge/rootchain"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestSAMPool_AddMessage(t *testing.T) {
+	t.Parallel()
+
+	t.Run(
+		"ErrInvalidHash",
+		func(t *testing.T) {
+			t.Parallel()
+
+			verifier := mockVerifier{
+				verifyHash: func(msg rootchain.SAM) error {
+					return errors.New("a really bad hash")
+				},
+			}
+
+			pool := New(verifier)
+
+			assert.ErrorIs(t,
+				pool.AddMessage(rootchain.SAM{}),
+				ErrInvalidHash,
+			)
+		},
+	)
```

Comment:
I think you need to make sure the pool is empty in case of error

Reply:
Deprecated due to responsibility being delegated to `samuel`